File: arc137/a.py
L,R = map(int,input().split())

def gcd(a,b):
    if b == 0:
        return a
    return gcd(b,a%b)

i_to = min(L+1500,R+1)
j_from = max(R-1500,L+1)
ans= -1
for i in range(L,i_to):
    for j in range(j_from,R+1):
        if j > i:
            if gcd(i,j) == 1:
                ans = max(ans,j-i)
            

print(ans)

File: arc137/a.cpp
#include <bits/stdc++.h>
using namespace std;

long long gcd(long long a, long long b) {
  if (b == 0)
    return a;
  return gcd(b, a % b);
}

int main() {
  long long L, R;
  cin >> L >> R;

  long long ans = 0;
  for (long long l = L; l <= min(R - 1, L + 1500); l++) {
    for (long long r = R; r >= max(L + 1, R - 1500); r--) {
    //   cout << l << " " << r << endl;
      if(gcd(l, r) == 1){
          ans = max(ans, r - l);
      }
    }
  }
  cout << ans << endl;
}